src/main/java/de/hysky/skyblocker/skyblock/waypoint/OrderedWaypoints.java

```diff
@@ -1,309 +1,90 @@
-
-		return Command.SINGLE_SUCCESS;
-	}
-
-	private static int showAll(FabricClientCommandSource source) {
-		source.sendFeedback(Constants.PREFIX.get().append(Text.translatable("skyblocker.waypoints.ordered.showAll")));
-		showAll = !showAll;
-
-		return Command.SINGLE_SUCCESS;
-	}
-
-	private static void render(WorldRenderContext wrc) {
-		if ((Utils.isInCrystalHollows() || Utils.isInDwarvenMines()) && loaded.isDone() && SEMAPHORE.tryAcquire()) {
-			for (OrderedWaypointGroup group : WAYPOINTS.values()) {
-				if (group.enabled()) {
-					List<OrderedWaypoint> waypoints = group.waypoints();
-					if (waypoints.isEmpty()) continue;
-
-					if (!showAll) {
-						ClientPlayerEntity player = MinecraftClient.getInstance().player;
-						int centreIndex = INDEX_STORE.computeIfAbsent(group.name(), name -> 0);
-
-						for (int i = 0; i < waypoints.size(); i++) {
-							OrderedWaypoint waypoint = waypoints.get(i);
-
-							if (waypoint.getPos().isWithinDistance(player.getPos(), RADIUS)) {
-								centreIndex = i;
-								INDEX_STORE.put(group.name(), i);
-
-								break;
-							}
-						}
-
-						int previousIndex = (centreIndex - 1 + waypoints.size()) % waypoints.size();
-						int currentIndex = (centreIndex + waypoints.size()) % waypoints.size();
-						int nextIndex = (centreIndex + 1) % waypoints.size();
-
-						OrderedWaypoint previous = waypoints.get(previousIndex);
-						OrderedWaypoint current = waypoints.get(currentIndex);
-						OrderedWaypoint next = waypoints.get(nextIndex);
-
-						previous.render(wrc, RelativeIndex.PREVIOUS, previousIndex);
-						current.render(wrc, RelativeIndex.CURRENT, currentIndex);
-						next.render(wrc, RelativeIndex.NEXT, nextIndex);
-
-						RenderHelper.renderLineFromCursor(wrc, Vec3d.ofCenter(next.getPos().up()), LIGHT_GRAY, 1f, 5f);
-					} else {
-						for (int i = 0; i < waypoints.size(); i++) {
-							//Render them as white by default
-							waypoints.get(i).render(wrc, RelativeIndex.CURRENT, i);
-						}
-					}
-				}
-			}
-
-			SEMAPHORE.release();
-		}
+												.executes(context -> fromSkyblockerFormat(context.getSource())))))));
 	}
 
+    /**
+     * Loads and migrates the ordered waypoints to waypoints.
+     * @deprecated Use {@link Waypoints} instead.
+     */
+    @Deprecated
 	private static void load() {
-		loaded = CompletableFuture.runAsync(() -> {
-			try (BufferedReader reader = Files.newBufferedReader(PATH)) {
-				WAYPOINTS.putAll(SERIALIZATION_CODEC.parse(JsonOps.INSTANCE, JsonParser.parseReader(reader)).getOrThrow());
-			} catch (NoSuchFileException ignored) {
-			} catch (Exception e) {
-				LOGGER.error("[Skyblocker Ordered Waypoints] Failed to load the waypoints! :(", e);
-			}
-		});
-	}
-
-	private static void save() {
-		try (BufferedWriter writer = Files.newBufferedWriter(PATH)) {
-			SkyblockerMod.GSON.toJson(SERIALIZATION_CODEC.encodeStart(JsonOps.INSTANCE, WAYPOINTS).getOrThrow(), writer);
+		try (BufferedReader reader = Files.newBufferedReader(PATH)) {
```

Comment:
Is there a reason for making this not blocking now that it doesn't really do much?

Reply:
This is so that `migrateOrderedWaypoints` is called synchronously to prevent race conditions with waypoints.